gateway/src/service.rs

```diff
@@ -263,6 +266,7 @@ impl GatewayService {
         let headers = req.headers_mut();
         headers.typed_insert(XShuttleAccountName(account_name.to_string()));
         headers.typed_insert(XShuttleAdminSecret(control_key));
+        headers.typed_insert(XShuttleProjectId(project_id));
```

Comment:
I'm not sure why we do this? I see there is a new arg on `deployer` which I expect does the same?

Reply:
Indeed. That was done previously to avoid sending the `project_id` as an arg to the deployer, and it remained like this although we went on the arg path. It is trivial now to use the `project_id` stored on the `Persistence` instance, attached as an extension, and avoid setting this header. Good catch! I will remove it since we don't need it anymore.